src/codeLens/prometheusScraper.ts

```diff
@@ -87,23 +97,43 @@ export class PrometheusCodeLensProvider implements vscode.CodeLensProvider {
         // Action lens
         this.codeLenses.push(
           new vscode.CodeLens(range, {
-            title: "Scrape data",
+            title: "Scrape data from endpoint",
             tooltip: "Connect to an exporter and scrape metrics, then use them in the Extension.",
             command: "dt-ext-copilot.codelens.scrapeMetrics",
             arguments: [],
           }),
         );
+        if (!this.fileScraped) {
+          this.codeLenses.push(
+            new vscode.CodeLens(range, {
+              title: "Scrape data from file",
+              tooltip: "Read a file, scrape metrics, then use them in the Extension.",
+              command: "dt-ext-copilot.codelens.scrapeMetricsFromFile",
+              arguments: [],
+            }),
+          );
+        }
         // Edit config lens
-        if (this.lastScrape !== "N/A") {
+        if (this.endpointScraped) {
           this.codeLenses.push(
             new vscode.CodeLens(range, {
-              title: "Edit config",
-              tooltip: "Make changes to the endpoint connection details",
+              title: "Edit endpoint config",
+              tooltip: "Make changes to the endpoint connection details.",
               command: "dt-ext-copilot.codelens.scrapeMetrics",
               arguments: [true],
             }),
           );
         }
+        if (this.fileScraped) {
+          this.codeLenses.push(
+            new vscode.CodeLens(range, {
+              title: "Scrape data from different file",
```

Comment:
Same. Don't push another lens.

One lens for "Edit config" is enough. What if I decide to change from file scraping to endpoint scraping? Can't do that this way.